### PR TITLE
CMS-1648 tree base routing helpers

### PR DESCRIPTION
**Pull request by sdonk**
To do (delete all that do not apply):

 - [X] Change has a jira ticket that has the correct status.
 - [X] Changelog entry added.

### CODE REVIEW

File: core/management/commands/enable_tree_based_routing.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+class Command(BaseCommand):
+    help = 'Enable the tree based routing flag on pages. ' \
+           'If --servicename is not specified, the flag is enabled on ' \
+           'ALL the pages.'
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--servicename',
+            help='Select international or domestic',
+            choices=['international', 'domestic']
```
Comment (richtier) [inline core/management/commands/enable_tree_based_routing.py]: https://docs.python.org/3/library/argparse.html#choices
> Any object that supports the in operator can be passed as the choices value, so dict objects, set objects, custom containers, etc. are all supported.
so I think you can pass `choices=SERVICES_MAPPING` and not do the dict lookup in `handle`

File: core/management/commands/enable_tree_based_routing.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        queryset = Page.objects.exclude(title__isnull=True)
+        if options['servicename']:
+            service_name = SERVICES_MAPPING[options['servicename']]
+            queryset = filters.ServiceNameFilter().filter_service_name(
+                queryset=queryset,
+                name=None,
+                value=service_name,
+            )
+        for page in queryset:
+            page.specific.uses_tree_based_routing = True
```
Comment (richtier) [inline core/management/commands/enable_tree_based_routing.py]: is `page.specific` cached? I have a suspicion that we need to do
```
page = page.specific
# do some mutation
page.save()
```

alternatively we can do `for page in queryset.specific():` and do it in one place

File: great_international/management/commands/set_missing_translated_titles.py [hunk 3]
```diff
@@ -37,11 +37,11 @@ def handle(self, *args, **options):
             value=cms.GREAT_INTERNATIONAL,
         )
         for page in queryset.specific():
-            print('--------------------------------------------------------')
-            print("{}: {} (ID:{})".format(
+            self.stdout.write('--------------------------------------------------------')
+            self.stdout.write('{}: {} (ID:{})'.format(
```
Comment (richtier) [inline great_international/management/commands/set_missing_translated_titles.py]: not a fan of f strings?
```
self.stdout.write(f'{page.__class__.__name__}: {page} (ID:{page.id})'
```

File: core/management/commands/enable_tree_based_routing.py (hunk 2 above)
Comment (sdonk) [reply core/management/commands/enable_tree_based_routing.py]: I tried `queryset.specific()` but it tries to call `Page` and the property `uses_tree_based_routing` doesn't exist on that class

File: great_international/management/commands/set_missing_translated_titles.py (hunk 3 above)
Comment (sdonk) [reply great_international/management/commands/set_missing_translated_titles.py]: I just replaced Andy's `print()` with `self.stdout.write`, as per Django's documentation.
I'll add the f strings

File: core/management/commands/enable_tree_based_routing.py (hunk 1 above)
Comment (sdonk) [reply core/management/commands/enable_tree_based_routing.py]: I can do `choices=SERVICES_MAPPING`and keep the lookup. If you pass a dict then either the keys or the values can be passed in the command, but it's easier to remember `domestic` than `EXPORT_READINESS` when you use the command

**Pull request merged**